domain/src/test/java/org/oppia/android/domain/platformparameter/PlatformParameterControllerDebugImplTest.kt

```diff
@@ -506,71 +662,102 @@ class PlatformParameterControllerDebugImplTest {
   }
 
   @Test
-  @Suppress("DeferredResultUnused")
-  fun testLoadParametersAsync_intParam_withRemoteAndNoLocalOverride_setsProcessStateToRemote() {
-    executeInPreviousAppInstance { testComponent ->
-      addTestRemotePlatformParameterToDatabase(
-        testComponent,
-        TEST_REMOTE_SYNC_UP_WORKER_PERIOD_HOURS
-      )
-      testComponent.getTestCoroutineDispatchers().runCurrent()
-    }
+  fun testDownloadRemoteParameters_returnsAsyncResultSuccess() {
     setUpTestApplicationComponent()
-    platformParameterControllerDebugImpl.loadParametersAsync()
-    testCoroutineDispatchers.runCurrent()
-    val platformParameterValueFromProcessState = platformParameterProcessState
-      .retrievePlatformParameterIntegerState(
-        PlatformParameterId.SYNC_UP_WORKER_TIME_PERIOD_IN_HOURS
-      )
-
-    assertThat(platformParameterValueFromProcessState)
-      .isEqualTo(TEST_REMOTE_SYNC_UP_WORKER_PERIOD_HOURS)
+    val downloadProvider = platformParameterControllerDebugImpl.downloadRemoteParameters()
+    val downloadMonitor = monitorFactory.createMonitor(downloadProvider)
+    val downloadResult = downloadMonitor.waitForNextResult()
+    assertThat(downloadResult).isInstanceOf(AsyncResult.Success::class.java)
   }
 
   @Test
-  @Suppress("DeferredResultUnused")
-  fun testLoadParametersAsync_withRemoteFlagAndNoLocalOverride_setsProcessStateToRemoteValue() {
-    TestPlatformParameterModule.forceEnableMultipleClassrooms(false)
-    executeInPreviousAppInstance { testComponent ->
-      addTestRemoteFeatureFlagToDatabase(testComponent, true)
-      testComponent.getTestCoroutineDispatchers().runCurrent()
-    }
+  fun testUpdateOverriddenFeatureFlags_returnsCorrectValue() {
```

Comment:
Here and below.
```suggestion
  fun testUpdateOverriddenFeatureFlags_withLocalOverride_updatesValue() {
```
As it's not actually returning the value here.

Reply:
updated